include/boost/beast/core/detail/impl/any_buffers.hpp

```diff
@@ -39,7 +38,13 @@ any_buffers<IsConst>::construct(Buffers const& b, Alloc const& a)
     auto alloc = allocator_type(a);
     using atraits = std::allocator_traits<allocator_type>;
 
-    auto p = atraits::allocate(a, sizeof(impl_type));
+    std::size_t extra = (std::min)(
+        static_cast<std::size_t>(
+            std::distance(net::buffer_sequence_begin(b),
+                          net::buffer_sequence_end(b))),
+            std::size_t(1)) - 1;
+
+    auto p = atraits::allocate(a, sizeof(impl_type) + sizeof(value_type) * extra);
```

Comment:
This formula calculates bytes, but the allocator type is measured in units of `sizeof(impl_type)` which means you are over-allocating by quite a lot.

Reply:
oh, that's annoying.

Reply:
```
    std::size_t extra = (std::min)(
        static_cast<std::size_t>(
            std::distance(net::buffer_sequence_begin(b),
                          net::buffer_sequence_end(b))),
        std::size_t(1)) - 1;
    
    const auto bytes = sizeof(impl_type) + sizeof(value_type) * extra;
    const auto units = (bytes + sizeof(impl_type) - 1) / sizeof(impl_type);
    auto p = atraits::allocate(a, units);
```